parse/IntComplexValueRefParser.cpp

```diff
@@ -102,6 +102,30 @@ namespace parse {
                     ) [ _val = new_<ValueRef::ComplexVariable<int> >(_a, _b, _c, _f, _d, _e) ]
                 ;
 
+            ship_parts_owned
+                =   (
+                            tok.ShipPartsOwned_ [ _a = construct<std::string>(_1) ]
+                        >-( parse::label(Empire_token) >    int_value_ref [ _b = _1 ] )
+                        >-( parse::label(Name_token)   >    string_value_ref [ _d = _1 ] )
+                    ) [ _val = new_<ValueRef::ComplexVariable<int>>(_a, _b, _c, _f, _d, _e) ]
+                ;
+
+            ship_part_class_owned
+                = (
+                            tok.ShipPartClassOwned_ [ _a = construct<std::string>(_1) ]
```

Comment:
probably don't need a separate name for this value in FOCS... could construct the parts of class owned variant when a class is specified, and the parts by name owned variant when a string is specified. Shouldn't be any ambiguity between these...